src/basic/fs-util.c

```diff
@@ -1135,3 +1140,53 @@ int xopenat(int dir_fd, const char *path, int flags, mode_t mode) {
 
         return TAKE_FD(fd);
 }
+
+int xopenat_lock(int dir_fd, const char *path, int flags, mode_t mode, LockType locktype, int operation) {
+        _cleanup_close_ int fd = -EBADF;
+        int r;
+
+        assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
+        assert(path);
+        assert(IN_SET(operation & ~LOCK_NB, LOCK_EX, LOCK_SH));
+
+        /* POSIX/UNPOSIX locks don't work on directories. */
```

Comment:
oh, they don't? i wasn't aware. bummer.

Reply:
LOCK_EX works on directories, see e.g. `strace -f -eflock -- flock . flock . true`.

Reply:
Hmm, these are fcntl locks, not flock.